### PR TITLE
Better test discovery

### PR DESCRIPTION
**Pull request by andy31415**
#### Problem
Searching for YAML files for figuring out what tests are valid assumes that chip-tool and source code are in sync (which is not always the case) and error prone (we have many yaml files).

#### Change overview
Execute chip-tool itself to figure out what tests are available

#### Testing
Validated that `./scripts/tests/run_test_suite.py list` is identical before and after

### CODE REVIEW

**Comment by andy31415** [comment]
fast track: test change only which should result in a noop.

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -87,9 +87,10 @@ jobs:
               timeout-minutes: 30
               run: |
                   ./scripts/run_in_build_env.sh \
-                  "./scripts/tests/run_test_suite.py run \
-                     --iterations 2 \
+                  "./scripts/tests/run_test_suite.py 
                      --chip-tool ./out/debug/standalone/chip-tool \
+                     run \
+                     --iterations 2 \
```
Comment (tcarmelveilleux) [inline .github/workflows/tests.yaml:93]: Won't this fail? `--chip-tool` is an argument to `run`.

File: scripts/tests/chiptest/__init__.py [hunk 2]
```diff
@@ -17,34 +17,38 @@
...
-        logging.debug('Found YAML: %s' % path)
 
-        # grab the name without the extension
-        name = path.stem
+    result = subprocess.run([chip_tool, 'tests'], capture_output=True)
 
-        if 'Simulated' in name:
+    # Unfortunately there is no 'LIST' command and result return code is 1 (i.e. error)
+    # however the above will return tests in stderr
+    test_re = re.compile('^  \\| \\* ([^ ]*) *\\|$')
```
Comment (tcarmelveilleux) [inline scripts/tests/chiptest/__init__.py:37]:
```suggestion
    test_re = re.compile(r'^  \| \* ([^ ]*) *\|$')
```

File: scripts/tests/chiptest/__init__.py [hunk 3]
```diff
@@ -17,34 +17,38 @@
...
-    for path in Path(os.path.join(root, 'src', 'app',  'tests', 'suites')).rglob("*.yaml"):
-        logging.debug('Found YAML: %s' % path)
 
-        # grab the name without the extension
-        name = path.stem
+    result = subprocess.run([chip_tool, 'tests'], capture_output=True)
 
-        if 'Simulated' in name:
+    # Unfortunately there is no 'LIST' command and result return code is 1 (i.e. error)
+    # however the above will return tests in stderr
```
Comment (vivien-apple) [inline scripts/tests/chiptest/__init__.py:36]: It should be pretty straightforward to add a list command if that makes your life simpler ?

Comment (vivien-apple) [reply scripts/tests/chiptest/__init__.py:36]: I just made #12322 for that.

**Review by vivien-apple — CHANGES_REQUESTED** [review]
Please use the `list` command. That should be less error-prone.

File: scripts/tests/chiptest/__init__.py (hunk 2 above)
Comment (andy31415) [reply scripts/tests/chiptest/__init__.py]: Logic updated, this change does not apply anymore.

File: scripts/tests/chiptest/__init__.py (hunk 3 above)
Comment (andy31415) [reply scripts/tests/chiptest/__init__.py]: done

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (andy31415) [reply .github/workflows/tests.yaml:93]: It is inconsistent, however not sure if  we can fix it in any other way as the commands depend on the tool (listing available tests).

I could place all paths  top level (and they would not be used for things like list). How strongly do you feel  about changing this?

File: scripts/tests/run_test_suite.py [hunk 4]
```diff
@@ -93,24 +94,28 @@ class RunContext:
...
     '--internal-inside-unshare',
     hidden=True,
     is_flag=True,
     default=False,
     help='Internal flag for running inside a unshared environment'
 )
+@click.option(
+    '--chip-tool',
+    default=FindBinaryPath('chip-tool'),
+    help='What chip tool app to use to run the test')
```
Comment (electrocucaracha) [inline scripts/tests/run_test_suite.py:108]: nit
```suggestion
    help='Binary path of chip tool app to use to run the test')
```

**Review by vivien-apple — APPROVED** [review]

**Review by jepenven-silabs — APPROVED** [review]

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (andy31415) [reply .github/workflows/tests.yaml:93]: Resolved: can followup if we think this is as important for user interaction.

**Pull request merged**